runtime/integration-tests/src/cases/liquidity_pools_gateway_queue.rs

```diff
@@ -1,41 +1,128 @@
+use cfg_primitives::currency_decimals;
 use cfg_traits::liquidity_pools::MessageQueue as MessageQueueT;
 use frame_support::assert_ok;
+use pallet_liquidity_pools::Message;
+use pallet_liquidity_pools_gateway::message::GatewayMessage;
+use runtime_common::account_conversion::AccountConverter;
 use sp_runtime::traits::One;
 
 use crate::{
+	cases::liquidity_pools::{
+		utils::*, AUSD_CURRENCY_ID, DEFAULT_BALANCE_GLMR, DEFAULT_DOMAIN_ADDRESS_MOONBEAM,
+		DOMAIN_MOONBEAM, GLMR_CURRENCY_ID, POOL_ID,
+	},
 	config::Runtime,
 	env::{Blocks, Env},
 	envs::fudge_env::{FudgeEnv, FudgeSupport},
-	utils::{currency::cfg, genesis, genesis::Genesis},
+	utils::{accounts::Keyring, currency::cfg, genesis, genesis::Genesis},
 };
 
-#[test_runtimes(all)]
-fn submit_and_process<T: Runtime + FudgeSupport>() {
-	let mut env = FudgeEnv::<T>::from_parachain_storage(
-		Genesis::default()
-			.add(genesis::balances::<T>(cfg(1_000)))
-			.storage(),
-	);
-
-	let expected_event = env.parachain_state_mut(|| {
-		let message = pallet_liquidity_pools::Message::AddPool { pool_id: 1 };
-		let nonce = <T as pallet_liquidity_pools_gateway_queue::Config>::MessageNonce::one();
-
-		assert_ok!(
-			<pallet_liquidity_pools_gateway_queue::Pallet<T> as MessageQueueT>::submit(
-				message.clone()
-			)
+mod inbound {
```

Comment:
The outbound part is already tested in most LP test but I'm thinking about adding some simple tests here as well. WDYT?

Reply:
Added below some comments regarding this

Reply:
I think it makes sense, at least one, to be sure the happy path is working correctly 👍🏻

Reply:
Indeed, I think it makes sense to have the test here checking the end-to-end with the router configuration. That way, we can test all `liquidity_pools.rs` tests avoiding the last event check and then remove all fudge usage from `liquidity_pools.rs` tests, because this test already ensures to be working correctly

Reply:
OK, I updated most integration tests around LP and dropped fudge where possible.